templates/_worker.js/routes-matcher.ts

```diff
@@ -496,7 +496,22 @@ export class RoutesMatcher {
 			return 'done';
 		}
 
-		const pathExistsInOutput = this.path in this.output;
+		let pathExistsInOutput = this.path in this.output;
+
+		// If a path with a trailing slash entered the `rewrite` phase and didn't find a match, it might
+		// be due to the `trailingSlash` setting in `next.config.js`. Therefore, we should remove the
+		// trailing slash and check again before entering the next phase.
+		if (
+			phase === 'rewrite' &&
+			!pathExistsInOutput &&
+			/^\/(.+)\/$/.test(this.path)
+		) {
+			const newPath = this.path.replace(/^\/(.+)\/$/, '/$1');
+			pathExistsInOutput = newPath in this.output;
+			if (pathExistsInOutput) {
+				this.path = newPath;
+			}
+		}
```

Comment:
sorry 😓 , last minute suggestion

Thinking about it I think that this code can be simplified a little?

```suggestion
		if (
			phase === 'rewrite' &&
			!pathExistsInOutput &&
			this.path.endsWith('/')
		) {
			const newPath = this.path.slice(0, -1);
			pathExistsInOutput = newPath in this.output;
			if (pathExistsInOutput) {
				this.path = newPath;
			}
		}
```
or:

```suggestion
		if (
			phase === 'rewrite' &&
			!pathExistsInOutput &&
			this.path.endsWith('/')
		) {
			const newPath = this.path.replace(/\/$/, '')
			pathExistsInOutput = newPath in this.output;
			if (pathExistsInOutput) {
				this.path = newPath;
			}
		}
```

?